commercetools-models/src/test/java/io/sphere/sdk/client/ApacheClientIntegrationTest.java

```diff
@@ -23,4 +25,23 @@ public void itWorks() throws Exception {
             assertThat(project.getKey()).isEqualTo(getSphereClientConfig().getProjectKey());
         }
     }
+
+
+    /**
+     * This Exception is caused by the fact that the client closes after trying api callback with invalid scope
+     * **/
+    @Test(expected = IllegalStateException.class )
+    public void stopRetriesOnInvalidConfig() throws Exception{
+        final SphereClientConfig clientConfig = getSphereClientConfig();
+        final SphereClientConfig badConfig = SphereClientConfig.of(clientConfig.getProjectKey()+"LL",clientConfig.getClientId(),clientConfig.getClientSecret() ,clientConfig.getAuthUrl() ,clientConfig.getApiUrl()  );
+        final HttpClient httpClient = newHttpClient();
+        final SphereAccessTokenSupplier tokenSupplier = SphereAccessTokenSupplier.ofAutoRefresh(badConfig, httpClient, false);
+        final SphereClient underlying = SphereClient.of(badConfig, httpClient, tokenSupplier);
+        BlockingSphereClient localClient = BlockingSphereClient.of(underlying, 30, TimeUnit.SECONDS);
+        try { assertProjectSettingsAreFine(localClient);} catch (Exception e) {}
+        //The sleep op here is added to avoid the race condition (the circuit breaker runs on a diff thread)
+        Thread.sleep(100);
```

Comment:
Does it make sense to use `assertEventually` here instead of the fixed `Thread.sleep`?

Reply:
actually here i want the exception catched by the test, this is why i didn't use the `assertEventually` here